src/components/Signup.tsx

```diff
@@ -0,0 +1,137 @@
+"use client";
+import React from "react";
+import { useFormik } from "formik";
+import axios from "axios";
+import { useRouter } from "next/navigation";
+
+const Signup = () => {
+  const router = useRouter();
+  const formik = useFormik({
+    initialValues: {
+      firstName: "",
+      lastName: "",
+      email: "",
+      fatherName: "",
+      motherName: "",
+      address: "",
+      pincode: "",
+      country: "INDIA",
+    },
+    onSubmit: async (values,{resetForm}) => {
+      try {
+        const res = await axios.post("/api/signup", values);
+        if (res.status === 201) {
+          router.push("/users");
+          resetForm();
+        }
+        console.log(values);
+        
+      } catch (error) {
+        console.log(error);
+      }
+    },
+  });
+
+  return (
+    <div className="bg-grey-lighter min-h-screen flex flex-col">
+      <div className="container max-w-sm mx-auto flex-1 flex flex-col items-center justify-center px-2">
+        <div className="bg-white px-6 py-8 rounded shadow-lg text-black w-full">
+          <h1 className="mb-5 text-3xl text-center">Sign Up</h1>
+          <form onSubmit={formik.handleSubmit}>
+            {/* first name--------------- */}
+            <input
```

Comment:
make this input more consize if its being used several times, create a common component and use map to render it.